### PR TITLE
[global] Add openapi spec for global values

### PR DESCRIPTION
**Pull request by name212**
## Description
Add global values openapi spec.
`global.modules.resourcesRequests.internal` move to `global.internal.modules.resourcesRequests`.
Many test changed for pass new validation. In most cases, we fixed `dockerRegistryCfg` from to base64 encoded `cfg` and added skipped field for `clusterConfiguration` section.
Because version 1.16 and 1.15 was removed, we fix [get_crds](https://github.com/deckhouse/deckhouse/pull/505/files#diff-22c2dc530cfec8d290ce4977e47c37ed4f19cff0ad0dbf7e872e7767df176eeb) test

## Why do we need it, and what problem does it solve?
Global configuration does not validate and apply without any checks.

## Changelog entries

```changes
module: global
type: feature
description: Add OpenAPI spec for global values. Validates configuration `global` section of a `d8-system/deckhouse` ConfigMap.
note: If Deckhouse upgrading on a new version was failed with the OpenAPI validation error, please report us. Do not change/remove params from d8-system/deckhouse ConfigMap, please wait for an OpenAPI specification fix. And you need to restart Deckhouse manually to apply updates because auto-updating begins work after global values validation. 
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "deploy/web/test"...

File: modules/810-deckhouse-web/templates/pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
+    helm.sh/hook-delete-policy: before-hook-creation
+  name: deckhouse-web
+  namespace: d8-system
+{{ include "helm_lib_module_labels" (list . (dict "app" "deckhouse-web")) | indent 2 }}
```
Comment (yalosev) [inline modules/810-deckhouse-web/templates/pdb.yaml:10]: As I remember we use `nindent` everywhere

Comment (name212) [reply modules/810-deckhouse-web/templates/pdb.yaml]: done

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/aws"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/openstack"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/openstack"...

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/openstack"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/openstack"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/openstack"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/azure"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/gce"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/gcp"...

**Review by yalosev — DISMISSED** [review]

File: global-hooks/openapi/values.yaml [hunk 2]
```diff
@@ -1,105 +1,347 @@
...
+              milliCpuMaster:
+                type: integer
+                format: int64
+                minimum: 0
+              memoryMaster:
+                type: integer
+                format: int64
+                minimum: 0
+  clusterConfiguration:
+    # TODO need to use relative path instead absolute
```
Comment (nabokihms) [inline global-hooks/openapi/values.yaml:47]: We are ok with using the `/deckhouse` path for now.

File: testing/openapi_validation/validators/enum.go [hunk 3]
```diff
@@ -57,6 +57,12 @@ var (
 			// vSphere
 			"apiVersions[0].openAPISpec.properties.cloud.properties.provider",
 		},
+		"global-hooks/openapi/values.yaml": {
+			"properties.clusterConfiguration.properties.apiVersion",
+			"properties.clusterConfiguration.properties.cloud.properties.provider",
+			"properties.deckhouseEdition",
```
Comment (nabokihms) [inline testing/openapi_validation/validators/enum.go:63]: Let's change enum to `[Unknown, CE, FE, EE ]`. The linter will be satisfied, and this change seems to affect only one hook.

File: modules/values.yaml [hunk 4]
```diff
@@ -1,7 +1,8 @@
 # Default global values section
 global:
-  discovery: {}
-  modulesImages: {}
+  # todo remove duplicate config values they should be in global-hooks/openapi/config-values.yaml only
+  # now we have strange behaviour in template tests
+  # probably, test helm render does not get defaults from global-hooks/openapi/config-values.yaml
```
Comment (nabokihms) [inline modules/values.yaml:5]: It sure does not, but it is about our tests code. Template tests when executed read files from various directories and merge them to get the result.

We solved the problem with template tests by moving values.yaml files to */template_tests directories for modules. Maybe there is a way to move this file somewhere too (or add the code to our tests which will read default global values from somewhere).

**Review by nabokihms — CHANGES_REQUESTED** [review]

**Review by z9r5 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/openstack"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/openstack"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "e2e/run/gcp"...

File: global-hooks/openapi/values.yaml (hunk 2 above)
Comment (name212) [reply global-hooks/openapi/values.yaml]: I removed comment

File: testing/openapi_validation/validators/enum.go (hunk 3 above)
Comment (name212) [reply testing/openapi_validation/validators/enum.go]: done

File: modules/values.yaml (hunk 4 above)
Comment (name212) [reply modules/values.yaml]: I stayed it as is.

**Review by z9r5 — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Pull request merged**